core/src/peer_id.rs

```diff
@@ -56,17 +52,9 @@ impl fmt::Display for PeerId {
 impl PeerId {
     /// Builds a `PeerId` from a public key.
     pub fn from_public_key(key: &PublicKey) -> PeerId {
-        let key_enc = key.to_protobuf_encoding();
-
-        let hash_algorithm = if key_enc.len() <= MAX_INLINE_KEY_LENGTH {
-            Code::Identity
-        } else {
-            Code::Sha2_256
-        };
-
-        let multihash = hash_algorithm.digest(&key_enc);
-
-        PeerId { multihash }
+        PeerId {
+            multihash: Code::Sha2_256.digest(&key.to_protobuf_encoding()),
```

Comment:
This will break lots of code without alerting maintainers through compile errors, i.e. this is really bad: Actyx for example only supports ed25519 and then makes use of the fact that the public key can be recovered from the PeerId.

Changing the bit format of PeerId (for the same key) also is a network breaking change, so such a proposal should come with a thorough analysis of the intended and unintended effects as well as a migration plan. And if the only benefit is to avoid a deprecated internal function, then I don’t think this is worth it.

Reply:
Thanks for commenting @rkuhn! There was a misunderstanding on the original issue that I've hopefully been able to clarify! :)

Reply:
Yes, thanks!